public/app/features/alerting/unified/components/rule-editor/GrafanaEvaluationBehavior.tsx

```diff
@@ -114,60 +219,369 @@ function FolderGroupAndEvaluationInterval({
+      <Stack direction="column" justify-content="flex-start" align-items="flex-start">
+        <Stack alignItems="center">
+          <div style={{ width: 420 }}>
+            <Field
+              label={label}
+              data-testid="group-picker"
+              className={styles.formInput}
+              error={errors.group?.message}
+              invalid={!!errors.group?.message}
+              htmlFor="group"
+            >
+              <Controller
+                render={({ field: { ref, ...field }, fieldState }) => (
+                  <AsyncSelect
+                    disabled={!folder || loading}
+                    inputId="group"
+                    key={uniqueId()}
+                    {...field}
+                    onChange={(group) => {
+                      field.onChange(group.label ?? '');
+                    }}
+                    isLoading={loading}
+                    invalid={Boolean(folder) && !group && Boolean(fieldState.error)}
+                    loadOptions={debouncedSearch}
+                    cacheOptions
+                    loadingMessage={'Loading groups...'}
+                    defaultValue={defaultGroupValue}
+                    defaultOptions={groupOptions}
+                    getOptionLabel={(option: SelectableValue<string>) => (
+                      <div>
+                        <span>{option.label}</span>
+                        {option.isProvisioned && (
+                          <>
+                            {' '}
+                            <ProvisioningBadge />
+                          </>
+                        )}
+                      </div>
                     )}
-                  </span>
+                    placeholder={'Select an evaluation group...'}
+                  />
                 )}
-              </Stack>
-            </div>
-          </Stack>
-        </div>
+                name="group"
+                control={control}
+                rules={{
+                  required: { value: true, message: 'Must enter a group name' },
+                }}
+              />
+            </Field>
+          </div>
+          <Box gap={1} display={'flex'} alignItems={'center'}>
+            <Text color="secondary">or</Text>
+            <Button
+              onClick={onOpenEvaluationGroupCreationModal}
+              type="button"
+              icon="plus"
+              fill="outline"
+              variant="secondary"
+              disabled={!folder}
+              data-testid={selectors.components.AlertRules.newEvaluationGroupButton}
+            >
+              <Trans i18nKey="alerting.rule-form.evaluation.new-group">New evaluation group</Trans>
+            </Button>
+          </Box>
+          {isCreatingEvaluationGroup && (
+            <EvaluationGroupCreationModal
+              onCreate={handleEvalGroupCreation}
+              onClose={() => setIsCreatingEvaluationGroup(false)}
+              groupfoldersForGrafana={groupfoldersForGrafana?.result}
```

Comment:
What happens if we display the modal but `groupfoldersForGrafana` is null? Should we prevent that?

Reply:
this is existing code (just moved)